pyeudiw/jwt/__init__.py

```diff
@@ -67,44 +103,56 @@ def encrypt(self, plain_dict: Union[dict, str, int, None], **kwargs) -> str:
         :returns: A string that represents the JWE.
         :rtype: str
         """
-        _key = key_from_jwk_dict(self.jwk.as_dict())
-
-        if isinstance(_key, cryptojwt.jwk.rsa.RSAKey):
-            JWE_CLASS = JWE_RSA
-        elif isinstance(_key, cryptojwt.jwk.ec.ECKey):
-            JWE_CLASS = JWE_EC
-        else:
-            raise JWEEncryptionError(
-                f"Error while encrypting: f{_key.__class__.__name__} not supported!")
-
-        _payload: str | int | bytes = ""
-
-        if isinstance(plain_dict, dict):
+        
+        jwe_strings =[]
+        
+        if isinstance(plain_dict,dict):
             _payload = json.dumps(plain_dict).encode()
         elif not plain_dict:
             _payload = ""
         elif isinstance(plain_dict, (str, int)):
             _payload = plain_dict
         else:
             _payload = ""
+            
+        for key in self.jwks:
+            if isinstance(key, cryptojwt.jwk.rsa.RSAKey):
+                JWE_CLASS = JWE_RSA
+            elif isinstance(key, cryptojwt.jwk.ec.ECKey):
+                JWE_CLASS = JWE_EC
+            else:
+                raise JWEEncryptionError(
+                    f"Error while encrypting: "
+                    f"{self.jwk.__class__.__name__} not supported!"
+                )
 
-        _keyobj = JWE_CLASS(
-            _payload,
-            alg=DEFAULT_ENC_ALG_MAP[_key.kty],
-            enc=DEFAULT_ENC_ENC_MAP[_key.kty],
-            kid=_key.kid,
-            **kwargs
-        )
-
-        if _key.kty == 'EC':
-            _keyobj: JWE_EC
-            cek, encrypted_key, iv, params, epk = _keyobj.enc_setup(
-                msg=_payload, key=_key)
-            kwargs = {"params": params, "cek": cek,
-                      "iv": iv, "encrypted_key": encrypted_key}
-            return _keyobj.encrypt(**kwargs)
-        else:
-            return _keyobj.encrypt(key=_key.public_key())
+            _keyobj = JWE_CLASS(
+                _payload,
+                alg = DEFAULT_ENC_ALG_MAP[key.kty],
+                enc = DEFAULT_ENC_ENC_MAP[key.kty],
+                kid = key.kid,
+                **kwargs
```

Comment:
Gusto mio personale: `kwargs` non documentati rendono il codice difficile da leggere e mantenere. Chi invoca questo metodo `encypt` non ha idea di cosa inserire nei kwargs (la risposta è che deve sapere come è fatta l'implementazione concreta di `cryptojwt.jwk` che viene effettivammente usata, che significa che questo latyer di isolamento in realtà è molto sottile).

Reply:
passa gli argomenti al backend cryptojwt, se questo supporta kwargs conviene mantenerlo